roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala

```diff
@@ -0,0 +1,28 @@
+package coop.rchain.rosette.prim
+
+import coop.rchain.rosette.{Ctxt, Ob, RblBool, Tuple}
+import coop.rchain.rosette.macros.{checkArgumentMismatch, checkTypeMismatch}
+import coop.rchain.rosette.prim.Prim._
+
+object tuple {
+  object tplCons extends Prim {
+    override val name: String = "tuple-cons"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Tuple] = {
+      val elem = ctxt.argvec.elem
+      checkTuple(1, elem).map(Tuple.cons(elem.head, _)) // arg1 must be a Tuple
+    }
+
+  }
+
+  private def checkTuple(n: Int, elem: Seq[Ob]): Either[PrimError, Tuple] =
```

Comment:
I see a lot of usage of `Either`, maybe worth considering is `Disjunction` from Scalaz? Much richer API plus (more importantly) right-biased by default

Reply:
Since 2.12.x `Either` is right-biased by default.

Reply:
Also, `cats.syntax.either` enriches the standard library `Either` API. 

(and makes it backward compatible with 2.11, though I doubt that matters in this project)

Reply:
@tymm was not aware, cool!
@henrytill scalaz vs cats, I honestly don't care as long we use on or the other :)

Reply:
@rabbitonweb agreed!

`roscala` seems to be using cats.  i'm using cats in `storage`.  `rholang`, OTOH, is using `scalaz`, so...